src/visualizer/vertex/Vertex.cpp

```diff
@@ -4,18 +4,61 @@
 
 #include "Vertex.hpp"
 
-namespace age::visualizer {
+namespace {
+using namespace age::visualizer;
+} // namespace
+
 Vertex::Vertex(int x, int y, QWidget* pParent) noexcept : QWidget(pParent) {
-  move(x - VERTEX_RADIUS / 2, y - VERTEX_RADIUS / 2);
-  resize(VERTEX_RADIUS + RESIZE_OFFSET, VERTEX_RADIUS + RESIZE_OFFSET);
-  setPalette(Qt::blue);
+  move(x - WIDGET_SIZE / 2, y - WIDGET_SIZE / 2);
+  resize(WIDGET_SIZE, WIDGET_SIZE);
+  setPalette(DEFAULT_VERTEX_COLOR);
 }
 
+Vertex::Vertex(QPoint const& point, QWidget* pParent) noexcept : Vertex(point.x(), point.y(), pParent) {}
+
 auto Vertex::paintEvent(QPaintEvent*) -> void {
   QPainter painter(this);
   painter.setPen(palette().color(backgroundRole()));
-  painter.drawEllipse(0, 0, VERTEX_RADIUS, VERTEX_RADIUS);
+  painter.drawEllipse((WIDGET_SIZE - DRAWN_VERTEX_DIAMETER) / 2, (WIDGET_SIZE - DRAWN_VERTEX_DIAMETER) / 2,
+                      DRAWN_VERTEX_DIAMETER, DRAWN_VERTEX_DIAMETER);
+}
+
+auto Vertex::mousePressEvent(QMouseEvent* pEvent) -> void {
+  if (distanceToCenter(pEvent->position().toPoint()) > DRAWN_VERTEX_DIAMETER / 2.0f) {
+    return;
+  }
+  switch (pEvent->button()) {
+    case Qt::LeftButton: handleLeftClickEvent(pEvent); break;
+    case Qt::RightButton: emit rightClickPressed(mapToParent(pEvent->position())); break;
```

Comment:
Best not to do one-liners in switch, easy to miss

```suggestion
    case Qt::LeftButton: 
      handleLeftClickEvent(pEvent); 
      break;
    case Qt::RightButton: 
      emit rightClickPressed(mapToParent(pEvent->position())); 
      break;
```

Reply:
Clang-format requires it to be a one-liner